articles/141_static_remapping.md

```diff
@@ -0,0 +1,435 @@
+
+- Node is in namespace `/ns` and uses relative name `bar`
+- User changes the default namespace to `/foo`
+- The final name is `/foo/bar`
+
+### Change the Node Name
+The node name is used in log messages and to create private names.
+ROS 1 has this feature using the argument `__node`.
+
+*Example:*
+
+- Node is named `camera_driver` and uses a private name `camera_info`
+- User changes the node name to `left_camera_driver`
+- The final name is `/ns/left_camera_driver/camera_info` and log messages use `left_camera_driver`
+
+## Remapping Names in ROS 1
+Remapping is a feature that also exists in ROS 1.
+In ROS 1 remapping works by passing in [arguments](http://wiki.ros.org/Remapping%20Arguments) to each node.
+Client libraries also have APIs in code to pass remapping rules when the node is initialized.
+A remap rule consists of two names: one that should be replaced with another.
+
+ROS 1 remapping works on **Fully Qualified Names** (FQN).
+Both sides of a rule are [expanded to FQN](http://docs.ros.org/api/roscpp/html/namespaceros_1_1names.html#a377ff8fede7b95398fd5d1c5cd49246b).
+Before a name is remapped it is also [expanded to FQN](http://docs.ros.org/api/roscpp/html/namespaceros_1_1names.html#ab2eebaf734abfbdccb4122f8e24f547f).
+The name is remapped to the right side only if it exactly matches the left side of a rule.
+
+
+## Remapping rule syntax
+This is a proposal for the ROS 2 remapping rule syntax.
+It attempts to be the same as ROS 1 syntax when possible.
+
+Use cases supported by this syntax:
+
+- Remap One Node in a Process
+- Change a Namespace
+- Change a Basename
+- Exact FQN Replacement
+- Exact Relative Name Replacement
+- Remap via Command Line
+- Change the Default Namespace
+- Change the Node Name
+
+Not supported:
+
+- Change a Token
+- Pre-FQN Remapping
+
+### How the Syntax Works
+
+The structure of a remapping rule is `match:=replacement`.
+`match` tests if a name should be remapped.
+`replacement` says what the new name will be.
+`:=` behaves the same as it does in ROS 1.
+
+Example rules are:
+
+- `foo:=bar`
+- `/foo/bar:=fiz/buzz`
+- `nodename:~/foo:=foo`
+- `**/foo:=\1/bar`
+
+#### Match Part of a Rule
+
+The match part of a rule uses these operators:
+
+- `*` matches a single token
+- `**` matches zero or more tokens delimited by slashes
+- `~` is replaced with the node's name anywhere it is used
+- `nodename:` prefixed to the match makes it apply only to a node with that name
+
+The operators `*` and `**` are similar to the globbing behavior in bash.
+`**` behaves similar to its use in bash>=4.0 with the globstar option set.
+
+The operators `~`, `*`, and `**` match whole tokens only.
+`*bar` looks like it would match `foobar`, but that would mean matching a partial token.
+To avoid confusion the operators are required to be separated from tokens by a `/`.
+For example `*/bar` is allowed, but `*bar` is invalid.
+This is incompatible with the ROS 1 syntax where `~bar` is used, but now `~/bar` must be used instead.
```

Comment:
This is also supported by the rules in the other document:

> - must separate a tilde (`~`) from the rest of the name with a forward slash (`/`), i.e. `~/foo` not `~foo`

-- http://design.ros2.org/articles/topic_and_service_names.html#ros-2-topic-and-service-name-constraints